### PR TITLE
Implemented method for adding random problems to participant

### PR DESCRIPTION
**Pull request by gogo4ds**
Add logic for giving set of random problems to participants in Online Contest
Closes https://github.com/SoftUni-Internal/suls-issues/issues/3711

### CODE REVIEW

File: Open Judge System/OJS.Services/Business/Participants/ParticipantsBusinessService.cs [hunk 1]
```diff
@@ -19,6 +20,7 @@ public class ParticipantsBusinessService : IParticipantsBusinessService
             this.participantsData = participantsData;
             this.contestsData = contestsData;
         }
+
```
Comment (jicata) [inline Open Judge System/OJS.Services/Business/Participants/ParticipantsBusinessService.cs]: Remove blank line

File: Open Judge System/OJS.Services/Business/Participants/IParticipantsBusinessService.cs [hunk 2]
```diff
@@ -7,6 +7,6 @@ public interface IParticipantsBusinessService : IService
     {
         bool CanCompeteByContestAndUserId(Contest contest, string userId);
 
-        Participant CreateNewByContestUserIdAndIsOfficial(Contest contest, string userId, bool isOfficial);
+        Participant CreateNewByContestUserIdIsOfficialAndIsAdmin(Contest contest, string userId, bool isOfficial, bool isAdmin);
```
Comment (jicata) [inline Open Judge System/OJS.Services/Business/Participants/IParticipantsBusinessService.cs]: This line is longer than 120 chars, split arguments on new lines

File: Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/CompeteController.cs [hunk 3]
```diff
@@ -100,8 +100,8 @@ public void ValidateContest(Contest contest, bool official)
         {
             if (contest == null ||
                 contest.IsDeleted ||
-                !contest.IsVisible ||
-                !this.IsUserAdminOrLecturerInContest(contest))
+                !(contest.IsVisible ||
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/CompeteController.cs]: Can you insert the negation (!) in the braces, the code is more readable when you don't have to negate a complex condition. It should be:
contest == null ||
contest.IsDeleted ||
(!contest.IsVisible &&
    !this.IsUserAdminOrLecturerInContest(contest)))

File: Open Judge System/OJS.Services/Business/Participants/ParticipantsBusinessService.cs [hunk 4]
```diff
@@ -35,24 +36,51 @@ public bool CanCompeteByContestAndUserId(Contest contest, string userId)
...
+                {
+                    this.AssignRandomProblemsToParticipant(participant, contest);
+                }
             }
             else
             {
                 participant = new Participant(contest.Id, userId, isOfficial);
             }
-
+
```
Comment (Innos) [inline Open Judge System/OJS.Services/Business/Participants/ParticipantsBusinessService.cs]: Delete emtpy space.

**Pull request merged**